aws-redis-cluster.yml

```diff
@@ -94,6 +94,16 @@ provision:
     details: Whether to enable encryption at rest.
     default: true
     prohibit_update: true
+  - field_name: data_tiering_enabled
+    type: boolean
+    details: |
+      Enables data tiering. Data tiering is only supported for replication groups using the r6gd node type.
+      This parameter must be set to true when using `r6gd` nodes.
+      Not all versions of Redis support this feature. Check supported versions.
+      For more information about Node Types and supported Redis versions see
+      https://docs.aws.amazon.com/AmazonElastiCache/latest/red-ug/CacheNodes.SupportedTypes.html
+    default: false
+    prohibit_update: true
```

Comment:
Prohibiting updating this field introduces unexpected behaviour:
- node_type field is undateable
- if node_type is set to an `r6gd` node_type but is later changed to a non `r6gd` node_type,
 data_tiering_enabled either (I don't know which will happen):
  - causes the node_type change to fail
  - node_type change succeeds but `data_tiering_enabled: true` no longer represents the reality

Reply:
@fnaranjo-vmw  The node_type change wouldn't succeed as you cannot move from a data_tiering enabled node type to a non-data-tiering node (and vice versa). So I think prohibiting the update in this field actually has the effect of providing rapid feedback to the user.

Reply:
Perfect. Everything looks nice then!
Great job @zucchinidev !

Reply:
Hey @pivotal-marcela-campo , do you know if we could encode this logic (changing from data-tiering to non-data tiering) as a terraform tests?
Or this logic only happens in the IAAS, and as such can't be expressed as a terraform test?